test/e2e/wasm/authz/authz_bench_integration_test.go

```diff
@@ -0,0 +1,172 @@
+package authz
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/bundle"
+	"github.com/open-policy-agent/opa/compile"
+	"github.com/open-policy-agent/opa/metrics"
+	"github.com/open-policy-agent/opa/storage"
+	"github.com/open-policy-agent/opa/test/authz"
+	testAuthz "github.com/open-policy-agent/opa/test/authz"
+	"github.com/open-policy-agent/opa/test/e2e"
+	"github.com/open-policy-agent/opa/util"
+	"github.com/open-policy-agent/opa/util/test"
+)
+
+var testRuntime *e2e.TestRuntime
+
+var queryPath = strings.Replace(authz.AllowQuery, ".", "/", -1)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+
+	testServerParams := e2e.NewAPIServerTestParams()
+
+	var err error
+	testRuntime, err = e2e.NewTestRuntime(testServerParams)
+	if err != nil {
+		os.Exit(1)
+	}
+
+	files := map[string]string{
+		"policy.rego": testAuthz.Policy,
+		".manifest":   `{"roots": ["/policy/restauthz"]}`,
+	}
+
+	bundleBuf := bytes.Buffer{}
+	ctx := context.Background()
+
+	// TODO: Is there an option to do this more easily?
+	// Would be cool to be able to pass already loaded data/files into the compile stuff
+	test.WithTempFS(files, func(rootDir string) {
+		err = compile.New().
+			WithAsBundle(true).
+			WithEntrypoints(strings.TrimPrefix(queryPath, "data/")). // Entrypoints shouldn't be "data" prefixed
+			WithOutput(&bundleBuf).
+			WithPaths(rootDir).
+			WithTarget(compile.TargetWasm).
+			Build(ctx)
+
+		if err != nil {
+			panic(err)
+		}
+	})
+
+	testBundle, err := bundle.NewCustomReader(bundle.NewTarballLoader(&bundleBuf)).Read()
+	if err != nil {
+		panic(err)
+	}
+
+	// Sneak the bundle in...
+	err = storage.Txn(ctx, testRuntime.Runtime.Store, storage.WriteParams, func(txn storage.Transaction) error {
+		compiler := ast.NewCompiler().WithPathConflictsCheck(storage.NonEmpty(ctx, testRuntime.Runtime.Store, txn))
+		m := metrics.New()
+
+		activation := &bundle.ActivateOpts{
+			Ctx:      ctx,
+			Store:    testRuntime.Runtime.Store,
+			Txn:      txn,
+			Compiler: compiler,
+			Metrics:  m,
+			Bundles:  map[string]*bundle.Bundle{"bundle1": &testBundle},
+		}
+
+		return bundle.Activate(activation)
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	errc := testRuntime.RunTests(m)
+	os.Exit(errc)
+}
+
+func BenchmarkRESTAuthzForbidAuthn(b *testing.B) {
+	runAuthzBenchmark(b, testAuthz.ForbidIdentity, 10)
+}
+
+func BenchmarkRESTAuthzForbidPath(b *testing.B) {
+	runAuthzBenchmark(b, testAuthz.ForbidPath, 10)
+}
+
+func BenchmarkRESTAuthzForbidMethod(b *testing.B) {
+	runAuthzBenchmark(b, testAuthz.ForbidMethod, 10)
+}
+
+func BenchmarkRESTAuthzAllow10Paths(b *testing.B) {
+	runAuthzBenchmark(b, testAuthz.Allow, 10)
+}
+
+func BenchmarkRESTAuthzAllow100Paths(b *testing.B) {
+	runAuthzBenchmark(b, testAuthz.Allow, 100)
+}
+
+func BenchmarkRESTAuthzAllow1000Paths(b *testing.B) {
+	runAuthzBenchmark(b, testAuthz.Allow, 1000)
+}
```

Comment:
It would be nice to include the output from go test -bench in the commit message for these so we have a rough sense of what we're looking at.

Reply:
Will do, ill add the output and a comparison against the similar test using the normal rego evaluations.